src/graphnet/models/data_representation/graphs/utils.py

```diff
@@ -486,6 +517,125 @@ def add_mean(
+    def add_time_first_pulse(
+        self,
+        time_index: int,
+        location: Optional[int] = None,
+    ) -> np.ndarray:
+        """Add the time of the first pulse.
+
+        Args:
+            time_index: Index of the time column in the padded tensor
+            location: Location to insert the time of the first pulse in the
+                    clustered tensor defaults to adding at the end
+        Altered:
+            clustered_x: The time of the first pulse is added at the end of
+                         the tensor or inserted at the specified location
+            _cluster_names: The names are added at the end of the tensor
+                            or inserted at the specified location
+        """
+        time_first_pulse = np.nanmin(self._padded_x[:, :, time_index], axis=1)
+        self._add_column(time_first_pulse, location)
+        # update the cluster names
+        if self._input_names is not None:
+            new_name = [self._input_names[time_index] + "_first_pulse"]
+            self._add_column_names(new_name, location)
+
+    def add_spread(
+        self, columns: List[int], location: Optional[int] = None
+    ) -> np.ndarray:
+        """Add the spread (max-min) of the columns."""
+        self._add_column(
+            np.nanmax(self._padded_x[:, :, columns], axis=1)
+            - np.nanmin(self._padded_x[:, :, columns], axis=1),
+            location,
+        )
+        # update the cluster names
+        if self._input_names is not None:
+            new_names = [self._input_names[i] + "_spread" for i in columns]
+            self._add_column_names(new_names, location)
+
+    def add_accumulated_value_after_t(
+        self,
+        time_index: int,
+        summarization_indices: List[int],
+        times: List[int],
+        location: Optional[int] = None,
+    ) -> np.ndarray:
+        """Add the cumulative sum of values after certain time.
+
+        NOTE: the time is counted from the first pulse of the sensor.
+        Make sure that the data is also sorted by time and be aware of
+        the standardization of the time column.
+
+        Args:
+            time_index: Index of the time column in the padded tensor
+            summarization_indices: List of column indices that defines
+                features that will be summarized with percentiles.
+            times: List of times after which the accumulated value
+                is calculated
+            location: Location to insert the accumulated values in the
+                      clustered tensor defaults to adding at the end
+        Altered:
+            clustered_x: The accumulated values are added at the end of
+                         the tensor or inserted at the specified location
+            _cluster_names: The names are added at the end of the tensor
+                            or inserted at the specified location
+        """
+        # Summarize the values at different times
+        time_first_pulse = np.nanmin(
```

Comment:
Could be made to a class feature since it is used in several of the class functions.